### PR TITLE
Add SSR experiments to a4a eid

### PR DESCRIPTION
**Pull request by jridgewell**
This adds the currently set SSR experiments to the a4a eid.

A small hiccup with this setup is that the SSR experiments are usually just boolean values (or small enums) that don't have a unique experiment id. I've created a pseudo id by concatenating the proto field id (eg, `25`) with a padded value (eg, `0`) to create the exp id (`2500`).

Re: #30504
Re: cl/335090010

### CODE REVIEW

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -2310,6 +2309,40 @@ export class AmpA4A extends AMP.BaseElement {
     }
     return null;
   }
+
+  /**
+   * Returns any enabled SSR experiments via the amp-usqp meta tag. These
+   * correspond to the proto field ids in cs/AmpTransformerParams.
+   *
+   * These experiments do not have a fully unique experiment id for each value,
+   * so we concatenate the key and value to generate a psuedo id. We assume
```
Comment (powerivq) [inline extensions/amp-a4a/0.1/amp-a4a.js:2318]: Should we create real exp id for these ssr features? I am not sure this pseudo id is permissible.

**Comment by jridgewell** [comment]
This is ready for review @powerivq @jeffkaufman

File: ads/google/a4a/utils.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ export const QQID_HEADER = 'X-QQID';
 export const SANDBOX_HEADER = 'amp-ff-sandbox';
 
 /**
- * Element attribute that stores experiment IDs.
+ * Element attribute that stores mendel experiment IDs.
```
Comment (lannka) [inline ads/google/a4a/utils.js:76]: Is mendel a valid concept here?

File: ads/google/a4a/utils.js [hunk 3]
```diff
@@ -199,6 +212,8 @@ export function googleBlockParameters(a4a, opt_experimentIds) {
     'ady': Math.round(slotRect.top),
     'oid': '2',
     'act': enclosingContainers.length ? enclosingContainers.join() : null,
+    // aexp URL param is separated by `!`, not `,`.
+    'aexp': aexps ? aexps.replace(/,/g, '!') : null,
```
Comment (lannka) [inline ads/google/a4a/utils.js:216]: is it a typo? aexp vs aexps?

**Review by powerivq — APPROVED** [review]

File: ads/google/a4a/utils.js (hunk 3 above)
Comment (jeffkaufman) [reply ads/google/a4a/utils.js:216]: not a typo, the param is called "aexp" internally.

Comment (lannka) [reply ads/google/a4a/utils.js:216]: ok, i suggest we make variable name the same to avoid mistakes.

File: ads/google/a4a/utils.js (hunk 2 above)
Comment (jridgewell) [reply ads/google/a4a/utils.js:76]: I'm not sure I understand?

File: ads/google/a4a/utils.js (hunk 3 above)
Comment (jridgewell) [reply ads/google/a4a/utils.js:216]: Done.

**Comment by jridgewell** [comment]
Can I get one more approval before merging? @lannka or @jeffkaufman?

**Review by lannka — APPROVED** [review]

**Review by jeffkaufman — APPROVED** [review]

**Pull request merged**